source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -764,14 +770,27 @@ ShaderGraphPtr ShaderGraph::create(const ShaderGraph* parent, const string& name
                         inputSocket->setGeomProp(geomprop->getName());
                         input->setGeomProp(geomprop->getName());
                     }
+
+                    const string& connection = nodedefPort->asA<Input>()->getOutputString();
+                    if (connection.empty())
```

Comment:
Is it really the nodedef port that should be checked for connection? I don't think nodedef's can't have connections. It's probably the node _instance_ port, `nodePort` above, that should be checked for connections.

It makes me wonder if this code segment is needed at all, since this condition would never have been true.. if I'm not missing something?

Reply:
Ah your right. I'm probably always adding in this now.

Reply:
This was always adding which wasn't correct. Does a check for an input now. Could have been it still worked since when we add upstream dependencies the default geom node would get replaced by whatever was upstream.